### ISSUE TITLE
Feature Request: Real-time type-ahead completions like zsh-autocomplete

### ISSUE THREAD

**Issue by naefl** [nushell/nushell#3784]
**Is your feature request related to a problem? Please describe.**
Discoverability of a new tool is key determinant on how much time I will invest in learning it. If it's easily discoverable, I will be much more likely to spend time learning it - and complexity is much more durable. `help` and Tab completions are great,  real-time type-ahead completions, similar to LSPs in editors (ideally fuzzy) are, however, next level wrt to discoverability. 

**Describe the solution you'd like**
Well, basically just this https://github.com/marlonrichert/zsh-autocomplete but in nushell

**Additional context**
Nothing but a deep sense of gratitude for such an awesome project :)

**Comment by rezural** [comment]
I've had a look at this before, and unfortunately it is highly non trivial. you basically need a sh environment running to perform the completions in shell land, that nu-shell can then use. This made me sad

**Comment by tristanCadet** [comment]
I totally agree on this, smart discoverability is a must have if we want Nushell to be a truly modern shell

-   it eliminates a lot of context switches (less googling) so it increases productivity and overall user experience
-   it makes learning faster and enables serendipity as options and docs are always on screen
-   it makes working with convoluted tools easier: e.g contextual auto-completion reduces the number of options to investigate


Some inspiration on what it could look like and how to implement it: [https://youtu.be/v2RmxZ9Vcps](https://youtu.be/v2RmxZ9Vcps)

This is the kind of feature that makes users stick with a tool. Smooth UX is a great thing to have to build a large community.
